misirlou/frontend/js/action-creators/search.js

```diff
@@ -92,6 +96,22 @@ const execSearch = debounce((query, dispatch, getSuggestions) =>
     }
 }, DEBOUNCE_INTERVAL);
 
+const searchAction = (query) => 
+{
+    return (dispatch, getState) =>
+    {
+        let start_state = getState().search.current;
+        Search.get(query).then(
+            response =>
+            {
+                if (start_state.query === getState().search.current.query)
```

Comment:
This check should occur in the error case as well. One way to do this would be to return the `getSearchAction` output from both promises, then to add a chained `.then` which does the check and the final dispatch.